In [1]:
import pandas as pd
from scipy import stats

In [8]:
aqi = pd.read_csv('../datasets/c4_epa_air_quality.csv')
aqi

Unnamed: 0  date_local            state_name           county_name  \
0             0  2018-01-01               Arizona              Maricopa   
1             1  2018-01-01                  Ohio               Belmont   
2             2  2018-01-01               Wyoming                 Teton   
3             3  2018-01-01          Pennsylvania          Philadelphia   
4             4  2018-01-01                  Iowa                  Polk   
..          ...         ...                   ...                   ...   
255         255  2018-01-01  District Of Columbia  District of Columbia   
256         256  2018-01-01             Wisconsin                 Dodge   
257         257  2018-01-01              Kentucky             Jefferson   
258         258  2018-01-01              Nebraska               Douglas   
259         259  2018-01-01        North Carolina                  Wake   

         city_name                                    local_site_name  \
0          Buckeye                                            BUCKEYE   
1        Shadyside                                          Shadyside   
2    Not in a city  Yellowstone National Park - Old Faithful Snow ...   
3     Philadelphia                             North East Waste (NEW)   
4       Des Moines                                          CARPENTER   
..             ...                                                ...   
255     Washington                                          Near Road   
256       Kekoskee                              HORICON WILDLIFE AREA   
257     Louisville                                       CANNONS LANE   
258          Omaha                                                NaN   
259  Not in a city                                         Triple Oak   

      parameter_name   units_of_measure  arithmetic_mean  aqi  
0    Carbon monoxide  Parts per million         0.473684    7  
1    Carbon monoxide  Parts per million         0.263158    5  
2    Carbon monoxide  Parts per million         0.111111    2  
3    Carbon monoxide  Parts per million         0.300000    3  
4    Carbon monoxide  Parts per million         0.215789    3  
..               ...                ...              ...  ...  
255  Carbon monoxide  Parts per million         0.244444    3  
256  Carbon monoxide  Parts per million         0.200000    2  
257  Carbon monoxide  Parts per million         0.163158    2  
258  Carbon monoxide  Parts per million         0.421053    9  
259  Carbon monoxide  Parts per million         0.188889    2  

[260 rows x 10 columns]

In [4]:
print("Using describe() to summarize AQI")
print(aqi.describe(include='all'))

print(aqi['state_name'].value_counts())


   Unnamed: 0  date_local    state_name   county_name      city_name  \
0           0  2018-01-01       Arizona      Maricopa        Buckeye   
1           1  2018-01-01          Ohio       Belmont      Shadyside   
2           2  2018-01-01       Wyoming         Teton  Not in a city   
3           3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4           4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158 

#### Hypothesis 1: ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.

In [14]:
ca_la = aqi[(aqi['county_name'] == 'Los Angeles')]
ca_la

Unnamed: 0  date_local  state_name  county_name         city_name  \
33           33  2018-01-01  California  Los Angeles         Lancaster   
42           42  2018-01-01  California  Los Angeles     Santa Clarita   
61           61  2018-01-01  California  Los Angeles          Pasadena   
76           76  2018-01-01  California  Los Angeles       Los Angeles   
109         109  2018-01-01  California  Los Angeles       Los Angeles   
110         110  2018-01-01  California  Los Angeles       Los Angeles   
119         119  2018-01-01  California  Los Angeles            Reseda   
132         132  2018-01-01  California  Los Angeles           Compton   
163         163  2018-01-01  California  Los Angeles             Azusa   
172         172  2018-01-01  California  Los Angeles       Pico Rivera   
177         177  2018-01-01  California  Los Angeles        Long Beach   
189         189  2018-01-01  California  Los Angeles            Pomona   
233         233  2018-01-01  California  Los Angeles          Glendora   
250         250  2018-01-01  California  Los Angeles  West Los Angeles   

                   local_site_name   parameter_name   units_of_measure  \
33       Lancaster-Division Street  Carbon monoxide  Parts per million   
42                   Santa Clarita  Carbon monoxide  Parts per million   
61                        Pasadena  Carbon monoxide  Parts per million   
76                    LAX Hastings  Carbon monoxide  Parts per million   
109  Los Angeles-North Main Street  Carbon monoxide  Parts per million   
110  Los Angeles-North Main Street  Carbon monoxide  Parts per million   
119                         Reseda  Carbon monoxide  Parts per million   
132                        Compton  Carbon monoxide  Parts per million   
163                          Azusa  Carbon monoxide  Parts per million   
172                 Pico Rivera #2  Carbon monoxide  Parts per million   
177            Long Beach (Hudson)  Carbon monoxide  Parts per million   
189                         Pomona  Carbon monoxide  Parts per million   
233                       Glendora  Carbon monoxide  Parts per million   
250               West Los Angeles  Carbon monoxide  Parts per million   

     arithmetic_mean  aqi  
33          0.394737    7  
42          0.394737    7  
61          0.789474   16  
76          0.863158   17  
109         0.994737   17  
110         0.900000   16  
119         1.015789   19  
132         1.742105   40  
163         0.673684   10  
172         1.047368   18  
177         1.200000   24  
189         1.000000   20  
233         0.389474    6  
250         0.655556   11

In [17]:
ca_other = aqi[(aqi['state_name'] == 'California') & ~(aqi['county_name'] == 'Los Angeles')]
ca_other

Unnamed: 0  date_local  state_name     county_name  \
16           16  2018-01-01  California  San Bernardino   
18           18  2018-01-01  California      Sacramento   
26           26  2018-01-01  California          Orange   
27           27  2018-01-01  California         Alameda   
34           34  2018-01-01  California          Fresno   
40           40  2018-01-01  California       San Mateo   
43           43  2018-01-01  California    Contra Costa   
45           45  2018-01-01  California           Butte   
46           46  2018-01-01  California       Riverside   
58           58  2018-01-01  California            Kern   
62           62  2018-01-01  California         Alameda   
63           63  2018-01-01  California      Sacramento   
75           75  2018-01-01  California     San Joaquin   
77           77  2018-01-01  California       Riverside   
81           81  2018-01-01  California   Santa Barbara   
86           86  2018-01-01  California     Santa Clara   
87           87  2018-01-01  California        Humboldt   
99           99  2018-01-01  California   Santa Barbara   
106         106  2018-01-01  California        Imperial   
117         117  2018-01-01  California       San Diego   
127         127  2018-01-01  California           Marin   
130         130  2018-01-01  California  San Bernardino   
134         134  2018-01-01  California         Alameda   
139         139  2018-01-01  California  San Bernardino   
141         141  2018-01-01  California   San Francisco   
147         147  2018-01-01  California     Santa Clara   
148         148  2018-01-01  California    Contra Costa   
151         151  2018-01-01  California          Orange   
156         156  2018-01-01  California          Fresno   
158         158  2018-01-01  California      Stanislaus   
161         161  2018-01-01  California            Kern   
170         170  2018-01-01  California       San Diego   
171         171  2018-01-01  California            Inyo   
176         176  2018-01-01  California    Contra Costa   
190         190  2018-01-01  California          Madera   
194         194  2018-01-01  California  San Bernardino   
198         198  2018-01-01  California       Riverside   
199         199  2018-01-01  California       Riverside   
200         200  2018-01-01  California       Riverside   
203         203  2018-01-01  California   Santa Barbara   
206         206  2018-01-01  California  San Bernardino   
211         211  2018-01-01  California          Fresno   
213         213  2018-01-01  California   Santa Barbara   
218         218  2018-01-01  California          Sonoma   
220         220  2018-01-01  California            Napa   
222         222  2018-01-01  California  San Bernardino   
224         224  2018-01-01  California      Sacramento   
227         227  2018-01-01  California   Santa Barbara   
230         230  2018-01-01  California        Monterey   
236         236  2018-01-01  California          Solano   
239         239  2018-01-01  California         Alameda   
249         249  2018-01-01  California          Orange   

                     city_name  \
16                     Ontario   
18                Arden-Arcade   
26                    La Habra   
27               Not in a city   
34                      Fresno   
40                Redwood City   
43                     Concord   
45                       Chico   
46                   Mira Loma   
58                       Arvin   
62                     Oakland   
63               Not in a city   
75                    Stockton   
77               Lake Elsinore   
81   Vandenberg Air Force Base   
86                    San Jose   
87                      Eureka   
99                      Goleta   
106                   Calexico   
117                   El Cajon   
127                 San Rafael   
130                    Fontana   
134                    Oakland   
139                Victorville   
141              San Franci

**Formulating the  null and alternative hypotheses:**

*   $H_0$: There is no difference in the mean AQI between Los Angeles County and the rest of California.
*   $H_A$: There is a difference in the mean AQI between Los Angeles County and the rest of California.

In [15]:
significance_lvl = 0.05

In [18]:
stats.ttest_ind(a=ca_la['aqi'], b=ca_other['aqi'], equal_var=False)

TtestResult(statistic=np.float64(2.1107010796372014), pvalue=np.float64(0.049839056842410995), df=np.float64(17.08246830361151))

###### With a p-value (0.049) being less than 0.05 (as the significance level is 5%), reject the null hypothesis in favor of the alternative hypothesis.

#### Hypothesis 2: With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?

In [19]:
ny = aqi[aqi['state_name'] == 'New York']
ohio = aqi[aqi['state_name'] == 'Ohio']

**Formulating the null and alternative hypotheses:**

*   $H_0$: The mean AQI of New York is greater than or equal to that of Ohio.
*   $H_A$: The mean AQI of New York is **below** that of Ohio.

In [21]:
stats.ttest_ind(a=ny['aqi'], b=ohio['aqi'], equal_var=False, alternative='less')

TtestResult(statistic=np.float64(-2.025951038880333), pvalue=np.float64(0.030446502691934697), df=np.float64(15.036745051598716))

###### p-value is less than the significance level. that's why rejecting the null hypothesis in favor of the alternative hypothesis

####  Hypothesis 3: A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?

In [23]:
michigan = aqi[aqi['state_name']=='Michigan']

**Formulating the null and alternative hypotheses here:**

*   $H_0$: The mean AQI of Michigan is less than or equal to 10.
*   $H_A$: The mean AQI of Michigan is greater than 10.


In [28]:
tstat, pvalue = stats.ttest_1samp(a=michigan['aqi'], popmean=10, alternative='greater')
print(tstat, pvalue)

-1.7395913343286131 0.9399405193140109


###### p-value(0.94) is greater than the significance level(0.05) that's why failed to reject the null hypothesis.